app/models/cookbook.rb

```diff
@@ -407,4 +401,23 @@ def deprecate_search(query)
   def copy_name_to_lowercase_name
     self.lowercase_name = name.to_s.downcase
   end
+
+  def create_new_collaborator(initiator)
+    unless collaborators.where(user_id: initiator.id).present?
```

Comment:
And `where(group_id: nil)`? This method will not create a new `Collaborator` if the `initiator` is a member of a collaborating group. That might be intentional, just checking.